schemes/rrtmgp/rrtmgp_inputs.F90

```diff
@@ -0,0 +1,655 @@
+   if (errflg /= 0) then
+      return
+   end if
+
+ end subroutine set_wavenumber_bands
+
+!=========================================================================================
+
+ subroutine get_band_index_by_value(swlw, targetvalue, units, nbnds, wavenumber_low, &
+                wavenumber_high, ans, errmsg, errflg)
+
+   ! Find band index for requested wavelength/wavenumber.
+
+   character(len=*),                      intent(in) :: swlw        ! sw or lw bands
+   real(kind_phys),                       intent(in) :: targetvalue  
+   character(len=*),                      intent(in) :: units       ! units of targetvalue
+   integer,                               intent(in) :: nbnds
+   real(kind_phys), target, dimension(:), intent(in) :: wavenumber_low
+   real(kind_phys), target, dimension(:), intent(in) :: wavenumber_high
+   character(len=*),                     intent(out) :: errmsg
+   integer,                              intent(out) :: errflg
+   integer,                              intent(out) :: ans
+
+   ! local
+   real(kind_phys), pointer, dimension(:) :: lowboundaries, highboundaries
+   real(kind_phys) :: tgt
+   integer  :: idx
+
+   character(len=*), parameter :: sub = 'get_band_index_by_value'
+   !----------------------------------------------------------------------------
+
+   ! Initialize error variables
+   errflg = 0
+   errmsg = ''
+   lowboundaries => wavenumber_low
+   highboundaries => wavenumber_high
+   if (trim(swlw) /= 'sw' .and. trim(swlw) /= 'lw') then
+      write(errmsg,'(a,a)') 'rrtmgp_inputs: get_band_index_by_value: type of bands not recognized: ', swlw
+      errflg = 1
+      return
+   end if
+
+   ! band info is in cm^-1 but target value may be other units,
+   ! so convert targetvalue to cm^-1
+   select case (units)
+   case ('inv_cm','cm^-1','cm-1')
+      tgt = targetvalue
+   case('m','meter','meters')
+      tgt = 1.0_kind_phys / (targetvalue * 1.e2_kind_phys)
+   case('nm','nanometer','nanometers')
+      tgt = 1.0_kind_phys / (targetvalue * 1.e-7_kind_phys)
+   case('um','micrometer','micrometers','micron','microns')
+      tgt = 1.0_kind_phys / (targetvalue * 1.e-4_kind_phys)
+   case('cm','centimeter','centimeters')
+      tgt = 1._kind_phys/targetvalue
+   case default
+      write(errmsg,'(a,a)') 'rrtmgp_inputs: get_band_index_by_value: units not recognized: ', units
+      errflg = 1
+   end select
+
+   ! now just loop through the array
+   ans = 0
+   do idx = 1,nbnds
+      if ((tgt > lowboundaries(idx)) .and. (tgt <= highboundaries(idx))) then
+         ans = idx
+         exit
+      end if
+   end do
+
+   if (ans == 0) then
+      write(errmsg,'(f10.3,a,a)') targetvalue, ' ', trim(units)
+      errflg = 1
+   end if
+   
+ end subroutine get_band_index_by_value
+
+ !=========================================================================================
+
+ pure logical function is_visible(wavenumber)
```

Comment:
I wonder if it would be worth moving this function into it's own file (maybe something like `atmos_phys_rad_utils`) under `phys_utils` and then writing a couple unit tests for it?  Of course this is arguably overkill, so if that is too annoying then feel free to just ignore this request or to make an issue so future us can worry about it instead.

Reply:
moved & unit tests added